src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngineUnitTest.java

```diff
@@ -0,0 +1,260 @@
+import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.*;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public final class VariantAnnotatorEngineUnitTest extends BaseTest {
+    @Test
+    public void testEmpty(){
+        final List<String> annotationGroupsToUse= Collections.emptyList();
+        final List<String> annotationsToUse= Collections.emptyList();
+        final List<String> annotationsToExclude= Collections.emptyList();
+        final FeatureInput<VariantContext> dbSNPBinding = null;
+        final VariantAnnotatorEngine vae = VariantAnnotatorEngine.ofSelectedMinusExcluded(annotationGroupsToUse, annotationsToUse, annotationsToExclude, dbSNPBinding);
+        Assert.assertTrue(vae.getGenotypeAnnotations().isEmpty());
+        Assert.assertTrue(vae.getInfoAnnotations().isEmpty());
+    }
+
+    @Test
+    public void testExclude(){
+        final List<String> annotationGroupsToUse= Collections.emptyList();
+        final List<String> annotationsToUse = Arrays.asList(Coverage.class.getSimpleName());//good one
+        final List<String> annotationsToExclude= annotationsToUse;
+        final FeatureInput<VariantContext> dbSNPBinding = null;
+        final VariantAnnotatorEngine vae = VariantAnnotatorEngine.ofSelectedMinusExcluded(annotationGroupsToUse, annotationsToUse, annotationsToExclude, dbSNPBinding);
+        Assert.assertTrue(vae.getGenotypeAnnotations().isEmpty());
+        Assert.assertTrue(vae.getInfoAnnotations().isEmpty());
+    }
+
+    @Test
+    public void testIncludeGroupExcludeIndividual(){
+        final List<String> annotationGroupsToUse= Collections.singletonList(StandardAnnotationKind.class.getSimpleName());
+        final List<String> annotationsToUse = Collections.emptyList();
+        final List<String> annotationsToExclude= Arrays.asList(Coverage.class.getSimpleName());
+        final FeatureInput<VariantContext> dbSNPBinding = null;
+        final VariantAnnotatorEngine vae = VariantAnnotatorEngine.ofSelectedMinusExcluded(annotationGroupsToUse, annotationsToUse, annotationsToExclude, dbSNPBinding);
+        Assert.assertFalse(vae.getGenotypeAnnotations().isEmpty());
+        Assert.assertFalse(vae.getInfoAnnotations().isEmpty());
+
+        //check that Coverage is out
+        Assert.assertTrue(vae.getInfoAnnotations().stream().noneMatch(a -> a.getClass().getSimpleName().equals(Coverage.class.getSimpleName())));
+    }
+
+    @Test
+    public void testAll(){
+        final List<String> annotationsToExclude= Collections.emptyList();
+        final FeatureInput<VariantContext> dbSNPBinding = null;
+        final VariantAnnotatorEngine vae = VariantAnnotatorEngine.ofAllMinusExcluded(annotationsToExclude, dbSNPBinding);
+        Assert.assertFalse(vae.getGenotypeAnnotations().isEmpty());
+        Assert.assertFalse(vae.getInfoAnnotations().isEmpty());
```

Comment:
Assert that the combined cardinality of these sets is equal to the number of annotation classes discovered by `Utils.makeInstances()`